### ISSUE TITLE
Build fails with autoconf 2.70 and up

### ISSUE THREAD

**Issue by mtolly** [deech/fltkhs#170]
On Manjaro with `autoconf` 2.71, attempting to build `fltkhs-hello-world` with `stack build --flag fltkhs:bundled`, I get:

```
fltkhs            > Running autoconf ...
fltkhs            > configure.ac:56: warning: The macro `AC_CONFIG_HEADER' is obsolete.
fltkhs            > configure.ac:56: You should run autoupdate.
fltkhs            > ./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
fltkhs            > configure.ac:56: the top level
fltkhs            > Warning: fltkhs.cabal:62:8: The field "build-tools" is deprecated in the Cabal
fltkhs            > specification version 2.0. Please use 'build-tool-depends' field
fltkhs            > Warning: fltkhs.cabal:60:6: The field "build-tools" is deprecated in the Cabal
fltkhs            > specification version 2.0. Please use 'build-tool-depends' field
fltkhs            > Configuring fltkhs-0.8.0.3...
fltkhs            > : WARNING: unrecognized options: --with-compiler
fltkhs            > : error: cannot find required auxiliary files: config.guess config.sub
```

https://lwn.net/Articles/839395/ contains this info:

```
*** More macros use config.sub and config.guess internally.

  As a consequence of improved support for cross compilation (see below),
  more macros now use the auxiliary scripts ‘config.sub’ and ‘config.guess’.
  If you use any of the affected macros, these scripts must be available
  when your configure script is run, even if you have no intention of
  ever cross-compiling your program.

  autoreconf will issue an error if any auxiliary scripts are needed but
  cannot be found.  (It is not currently possible to make autoconf
  itself issue this error.)

  ‘autoreconf --install’ will add ‘config.sub’, ‘config.guess’, and
  ‘install-sh’ to your source tree if they are needed.  If you are
  using Automake, scripts added to your tree by ‘autoreconf --install’
  will automatically be included in the tarball produced by ‘make dist’;
  otherwise, you will need to arrange for them to be distributed
  yourself.

  See the “Input” section of the manual for more detail, including
  where to get the auxiliary scripts that may be needed by autoconf macros.
```

**Comment by deech** [comment]
Last summer I moved `fltkhs` off `autotools` and to `cmake` and consolidated the various demo repos and utilities into https://github.com/deech/fltkhs/tree/TheGreatConsolidation. Also it works much better with recent Cabal versions but I never got around to merging any of it to `master`. Would you be up for trying it out? I haven't changed the API so hopefully it won't break your app. Also the `bundled` flag now defaults to `True` so just `stack build` should work.

**Comment by deech** [comment]
Thanks for the report!

**Comment by deech** [comment]
Last summer I cleaned up a ton of build issues, made it work well with `cabal` and pulled together all the various demo and utility repos into https://github.com/deech/fltkhs/tree/TheGreatConsolidation, but I never got around to merging it to `master`. I have fixed the `autoconf` issue, upgraded the bundled `fltk` and addressed a couple other things in that branch. Would you be up for trying it out? There should be no API breakage. The `bundled` flag is now on by default so you can do `stack build fltkhs`. Note that you have to explicitly specify `fltkhs` in the `stack build` command otherwise it will try to build everything and you will need to delete `/tmp/fltkhs-0.8.0.3-third-party-build/` so it re-compiles the new bundled `fltk`.

**Comment by mtolly** [comment]
I gave it a try in my app, and it appears to work great on Windows. However on Manjaro I get this:

```
> Linking .stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.1.0/build/onyx/onyx ...
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/bindings-opengl/c-lib/libfltkc.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk_images.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk_jpeg.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk_png.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk_z.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk_gl.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk_forms.a
> /usr/bin/ld: cannot find -l:/tmp/fltkhs-0.8.0.3-third-party-build/fltk-bundled-opengl/lib/libfltk.a
> collect2: error: ld returned 1 exit status
> `g++' failed in phase `Linker'. (Exit code: 1)
```

Using Stackage `lts-17.12`. The files do of course exist at these locations. I thought the colons before the filenames might be the issue, but I tried patching `Setup.hs` (removing the colon from [this line](https://github.com/deech/fltkhs/blob/272d225e641e9b6161df82066e9a4b919c0e812a/fltkhs/Setup.hs#L334)) and it had the same result, just without the colons. This error doesn't seem to exist in `fltkhs-demos` however, I can compile those fine. So there might just be something odd in my project, I'll keep investigating.

**Comment by deech** [comment]
If your project is OS do you mind posting a link?

**Comment by mtolly** [comment]
Figured it out, it was just these options that I had copied from the old version of `fltkhs-hello-world`:

```
ghc-options: -pgml g++ "-optl-Wl,--allow-multiple-definition" "-optl-Wl,--whole-archive" "-optl-Wl,-Bstatic" "-optl-Wl,-lfltkc" "-optl-Wl,-Bdynamic" "-optl-Wl,--no-whole-archive"
```

Removing all that and doing a full rebuild of the snapshot (plus deleting the `/tmp` subfolder) worked.

My app is [here](https://github.com/mtolly/onyxite-customs/tree/master/haskell), it is fairly large with a bunch of other dependencies.

**Comment by deech** [comment]
Awesome, thanks for letting me know. You shouldn't need those options in the future.

**Comment by deech** [comment]
Cool app too. Thanks for the link.

**Comment by gelisam** [comment]
For those who, like me, found this thread after encountering the `cannot find required auxiliary files: config.guess config.sub` error above, but then encountered this different error after switching to the branch [TheGreatConsolidation](https://github.com/deech/fltkhs/tree/TheGreatConsolidation):

```
$ git clone git@github.com:deech/fltkhs.git
$ cd fltkhs
$ git checkout TheGreatConsolidation
$ stack build --flag fltkhs:bundled
[...]
fltkhs                   > make[1]: Leaving directory '/home/gelisam/working/haskell/fltkhs/fltkhs/c-src'
fltkhs                   > Preprocessing library for fltkhs-0.8.0.3..
fltkhs                   > c2hs: C header contains errors:
fltkhs                   > 
fltkhs                   > .stack-work/dist/x86_64-linux-tinfo6/Cabal-3.2.0.0/build/Graphics/UI/FLTK/LowLevel/Base/Adjuster.i:1: (column 1) [ERROR]  >>> Lexical error !
fltkhs                   >   The character '#' does not fit here.
fltkhs                   >
[...]
```

This second error is discussed [here](https://github.com/haskell/c2hs/issues/268) and can be fixed by adding `c2hs-0.28.8` and `language-c-0.9` to the `extra-deps` section of `stack.yaml`.
